server/src/test/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinatorTest.java

```diff
@@ -1202,6 +1238,70 @@ public void testUsedHugeTimeRangeTrickyFilter2() throws IOException
     );
   }
 
+
+  @Test
+  public void testEternitySegmentWithStringComparison() throws IOException
+  {
+    coordinator.announceHistoricalSegments(
+        ImmutableSet.of(
+            eternitySegment
```

Comment:
We should also add a test case with segments with eternity and a segment with 2020-Month. 
Similar comment for other test cases as well.

Reply:
Added